src/client/components/pages/entities/cbReviewModal.tsx

```diff
@@ -0,0 +1,549 @@
+				type: 'danger'
+			}
+		});
+	};
+
+	getAccessToken = async () => {
+		try {
+			const response = await request
+				.post('/external-service/critiquebrainz/refresh');
+
+			if (response?.status === 200 && response?.body?.accessToken) {
+				return response.body.accessToken;
+			}
+
+			return null;
+		}
+		catch (error) {
+			this.handleError(error, 'Error');
+		}
+		return null;
+	};
+
+	handleInputChange = (
+		event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+	) => {
+		const {target} = event;
+		const value =
+			target.type === 'checkbox' ?
+				(target as HTMLInputElement).checked :
+				target.value;
+		const {name} = target;
+
+		this.setState({
+			[name]: value
+		} as any);
+	};
+
+	handleTextInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+		const {reviewValidateAlert} = this.state;
+		event.preventDefault();
+		// remove excessive line breaks to match formatting to CritiqueBrainz
+		const input = event.target.value.replace(/\n\s*\n\s*\n/g, '\n');
+		if (input.length <= this.maxTextLength) {
+			// cap input at maxTextLength
+			this.setState({textContent: input});
+		}
+
+		if (reviewValidateAlert && input.length >= this.minTextLength) {
+			// if warning was shown, rehide it when the input meets minTextLength
+			this.setState({
+				reviewValidateAlert: null
+			});
+		}
+	};
+
+	handleRatingsChange = (rate: number) => {
+		this.setState({
+			rating: rate / 20
+		});
+	};
+
+	resetCBReviewForm = () => {
+		this.setState({
+			acceptLicense: false,
+			rating: 0,
+			reviewValidateAlert: null,
+			success: true,
+			textContent: ''
+		});
+	};
+
+	handleCloseModal = () => {
+		this.setState({
+			acceptLicense: false,
+			alert: {
+				message: '',
+				title: '',
+				type: ''
+			},
+			language: 'en',
+			rating: 0,
+			reviewValidateAlert: null,
+			textContent: ''
+		});
+		this.props.handleModalToggle();
+	};
+
+	handleSubmitToCB = async (
+		event?: React.FormEvent<HTMLFormElement>
+	): Promise<null> => {
+		if (event) {
+			event.preventDefault();
+		}
+
+		const {
+			entityBBID,
+			entityType,
+			userId
+		} = this.props;
+
+		const {
+			acceptLicense,
+			language,
+			rating,
+			textContent
+		} = this.state;
+
+		if (textContent.length < this.minTextLength) {
+			this.setState({
+				reviewValidateAlert: `Your review needs to be longer than ${this.minTextLength} characters.`
+			});
+			return null;
+		}
+
+		if (userId &&
+			this.accessToken &&
+			acceptLicense
+		) {
+			let nonZeroRating: number;
+			if (rating !== 0) {
+				nonZeroRating = rating;
+			}
+
+			const review = {
+				entityBBID,
+				entityType,
+				language,
+				rating: nonZeroRating,
+				textContent
+			};
+
+			try {
+				let result: any;
+				const response = await request.post(`/${entityType}/${entityBBID}/reviews`)
+					.set('Content-Type', 'application/json')
+					.send({
+						accessToken: this.accessToken,
+						review
+					});
+
+				if (response.status >= 200 && response.status < 300) {
```

Comment:
Fetch already does that neatly for us :)
```suggestion
				if (response.ok) {
```